template/deployments/kustomize/draft.yaml

```diff
@@ -85,6 +97,68 @@ variables:
       value: "1Gi"
     description: "resource request for Memory"
     versions: ">=0.0.1"
+  - name: "PROBETYPE"
+    type: "string"
+    kind: "kubernetesProbeType"
+    default:
+      disablePrompt: true
+      value: "tcpSocket"
+    description: "The type of probe to use for container probes. Options are httpGet or tcpSocket"
+    versions: ">=0.0.1"
+    allowedValues:
+      - "httpGet"
+      - "tcpSocket"
+  - name: "PROBEHTTPPATH"
+    type: "string"
+    kind: "kubernetesProbeHttpPath"
+    default:
+      disablePrompt: true
+      value: "/"
+    conditionalReference:
+      variable: "PROBETYPE"
+      conditionValue: "httpGet"
+    description: "The path to use for the httpGet probes"
```

Comment:
`activeWhen`?

Reply:
as an array of conditions

Reply:
we should check that the value is in the alloweValues if they exist on the variableCondition

Reply:
swapped the naming, added the logic for checking variableConditions in the UT's, and added checks to Default handling and prompting!